### ISSUE TITLE
Toggle Component: Improve accessibility 

### ISSUE THREAD

**Issue by jbadan** [SAP/fundamental-react#321]
### Description

The Toggle component is not accessible in it's current state. 

Needs:
* Keyboard handling (press "enter" to toggle)
* label validation - if no label provided, add hidden label for screen readers.
* `aria-checked` attribute

### Versions
**fundamental-react:** 0.3.0.rc-2
**fiori-fundamentals:**

---

_**NOTE:** Where applicable, please include uncropped screen captures._

_**DISCLAIMER:**
After triaging an issue, the fundamental-react team will see if it can be reproduced or confirmed. If more information is needed, the fundamental-react team will contact the author.  Any issues awaiting responses from the author for more than 7 days will be closed.  The author can re-open the issue at a later time if they can present the requested information._

**Comment by singh-mahendra** [comment]
@jbadan Is this still open?
Not able to find a Toggle component in the src.Is this the Switch Component?
Also, do we have any accessibility guideline for the project that we can refer?

**Comment by jbadan** [comment]
Good questions! Toggle is now Switch component. It looks like we addressed the `aria-checked` aspect of this, but it is still missing:
1. keyboard handling (press enter to toggle, currently only works with space)
1. `role="switch"`  on `input`  (needs to be added to fundamental-styles documentation as well)
1. hidden label - this is where I'm not sure if we meant adding an extra `<label>` with a `screen-reader-only` css class or if we just needed `aria-label` on the `input`. @jacobdevera @prsdthkr @meganaconley  do you happen to remember?

**Comment by prsdthkr** [comment]
@jbadan I guess having the `aria-label` set on the `input` is better. That way, we don't have to add, associate and style another `label` element.

**Comment by singh-mahendra** [comment]
@jbadan Should the aria-label be mandatory,or can we default it to some value? say 'Toggle'.

**Comment by jbadan** [comment]
I think we can set the default for our playground pages - but the custom proptype we use for localization will make it required: https://github.com/SAP/fundamental-react/blob/master/src/utils/CustomPropTypes/CustomPropTypes.js#L78 (seen used here in Counter: https://github.com/SAP/fundamental-react/blob/e3e6c9ae7986085a4ce65b39d2da853db258fd06/src/Counter/Counter.js#L36)
